lib/ansible/modules/cloud/amazon/cloudfront_origin_access_identity.py

```diff
@@ -0,0 +1,249 @@
+
+    def update_origin_access_identity(self, caller_reference, comment, origin_access_identity_id, e_tag):
+        try:
+            return self.client.update_cloud_front_origin_access_identity(CloudFrontOriginAccessIdentityConfig={
+                    'CallerReference': caller_reference,
+                    'Comment': comment
+                },
+                Id=origin_access_identity_id, IfMatch=e_tag)
+        except (ClientError, BotoCoreError) as e:
+               self.module.fail_json_aws(e, msg="Error updating Origin Access Identity.") 
+
+
+class CloudFrontOriginAccessIdentityValidationManager(object):
+    """
+    Manages Cloudfront Origin Access Identities
+    """
+
+    def __init__(self, module):
+        self.module = module
+        self.__cloudfront_facts_mgr = CloudFrontFactsServiceManager(module)
+
+    def validate_etag_from_origin_access_identity_id(self, origin_access_identity_id):
+        try:
+            if origin_access_identity_id is None:
+                return
+            oai = self.__cloudfront_facts_mgr.get_origin_access_identity(origin_access_identity_id)
+            if oai is not None:
+                return oai.get('ETag')
+        except (ClientError, BotoCoreError) as e:
+               self.module.fail_json_aws(e, msg="Error getting etag from origin_access_identity.") 
+
+    def validate_origin_access_identity_id_from_caller_reference(
+            self, caller_reference):
+        try:
+            origin_access_identities = self.__cloudfront_facts_mgr.list_origin_access_identities()
+            origin_origin_access_identity_ids = [oai.get('Id') for oai in origin_access_identities]
+            for origin_access_identity_id in origin_origin_access_identity_ids:
+                oai_config = (self.__cloudfront_facts_mgr.get_origin_access_identity_config(origin_access_identity_id))
+                temp_caller_reference = oai_config.get('CloudFrontOriginAccessIdentityConfig').get('CallerReference')
+                if temp_caller_reference == caller_reference:
+                    return origin_access_identity_id
+        except (ClientError, BotoCoreError) as e:
+               self.module.fail_json_aws(e, msg="Error getting Origin Access Identity from caller_reference.") 
+
+    def validate_comment(self, comment):
+        if comment is None:
+            return "origin access identity created by Ansible with datetime " + datetime.datetime.now().strftime('%Y-%m-%dT%H:%M:%S.%f')
+        return comment
+
+
+def main():
+    argument_spec = ec2_argument_spec()
+
+    argument_spec.update(dict(
+        state=dict(choices=['present', 'absent'], default='present'),
+        origin_access_identity_id=dict(),
+        caller_reference=dict(),
+        comment=dict(),
+    ))
+
+    result = {}
+    e_tag = None
+    changed = False
+
+    module = AnsibleAWSModule(argument_spec=argument_spec, supports_check_mode=False)
```

Comment:
If checkmode isn't supported please add this in the 
`notes:` section in `DOCUMENTATION`

Reply:
have added to documentation under notes: section. let me know if this is inforrect

Reply:
Looks good, thanks

Reply:
oh, appologies, I think this needs to be a list